### PR TITLE
Update Dask content

### PR DESCRIPTION
**Pull request by mcfarljm**
Clarified the introduction section, added installation section, updated the code examples, and rearranged the order (move dask-jobqueue in front of dask-mpi).  I tested all of the examples, but dask-mpi gave me some errors.  It looks like it may be related to some known issues on GitHub, so I added a "warning" block about this.

### CODE REVIEW

File: docs/Documentation/Development/Languages/Python/dask.md [hunk 1]
```diff
@@ -9,172 +9,192 @@ grand_parent: Programming Languages
...
+    #SBATCH --partition=debug
+    
+    ml intel-mpi
+    mpiexec -np 4 python dask_mpi_example.py
+    ```
+    
+The job is then launched as:
+
+```bash
+sbatch dask_mpi_launcher.sh
```
Comment (k1nshuk) [inline docs/Documentation/Development/Languages/Python/dask.md:188]: An example combining `dask-jobqueue` and `dask-mpi` on the HPC could be useful. That would help give users some context as to when to use which tool and how to combine them.

**Review by k1nshuk — APPROVED** [review]
Overall this looks good for the time being. But we should definitely look at expanding this with more examples at a later date.

File: docs/Documentation/Development/Languages/Python/dask.md (hunk 1 above)
Comment (k1nshuk) [reply docs/Documentation/Development/Languages/Python/dask.md:188]: I haven't used `SLURMCluster` but the changes needed to run multi-node jobs seem about the same as those needed for dask-mpi. Digging deeper,  I agree that `dask-jobqueue` and `dask-mpi` cannot be combined in a single run. 

I did have a question though, what happens when you run `python dask_slurm_example.py` and you log out of the login node. Does the job continue to run while the user is "offline"?

Comment (mcfarljm) [reply docs/Documentation/Development/Languages/Python/dask.md:188]: If you need to log out you can use tmux or nohup to background the process.  But if you kill the main Python process, that will kill the scheduler.  It is supposed to send a signal to stop the workers, and apparently they also kill themselves automatically after 60 seconds with a non-existent scheduler.

**Pull request merged**